pages/docs/tutorials/native/apple-framework.md

```diff
@@ -0,0 +1,284 @@
+    NSString* ret = [Demo stringsStr:(@"That is string")];
+    return 0;
+}
+```
+</div>
+
+We call Kotlin classes directly from Objective-C code. Kotlin `object` has the class method 
+function `demoObject`, which allows us to get the only instance of the object and to call 
+`DemoObject` methods on it. 
+The standard pattern is used to create an instance of the `DemoClazz` class. We call
+the `[[ DemoDemoClazz alloc] init]` on Objective-C.
+Global functions from the Kotlin sources are scoped under the `Demo` class on Objective-C.
+All methods are turned into class methods of that class.
+The `strings` function is turned into `Demo.stringsStr` function in Objective-C, we are
+able to pass `NSString` directly to it. The return is visible as `NSString` too.
+
+## Using the Code from Swift
+
+The Framework, that we compiled with Kotlin/Native, has helper attributes to make it
+easier to with it from Swift. Let's convert the previous Objective-C example
+into Swift. As the result we'll have the following code in `main.swift`:
+
+<div class="sample" markdown="1" mode="swift" theme="idea" data-highlight-only="1" auto-indent="false">
+
+```swift
+let field = DemoDemoObject().field
+
+let clazz = DemoDemoClazz()
+clazz.memberFunction(p: 42)
+
+Demo.forIntegers(b: 1, s: 2, i: 3, l: 4)
+
+let ret = Demo.strings(str: "That is string")
+``` 
+</div>
+
+The Kotlin code is turned into nearly the similar looking
+code in Swift. There are some small differences, so far. We create the Kotlin `object DemoObject`
+as a class in Swift to access it's methods. We may see from the Objective-C attributes 
+in `<Framework>/Headers/Demo.h` that it is the way to access the object. 
+Methods and property names are translated as-is. Kotlin `String` is turned into Swift `String` too.
+
+# XCode and Framework Dependencies
+
+It may turn out tricky to configure the Objective-C or Swift project in XCode or
+JetBrains AppCode. You should include the Framework into the *General* tab of Target settings
+of XCode project. Also, you may need to fix the `@rpath` to include our Framework folder in the 
+search path, for that
+just search for *Runpath Search Paths* in the *Build Settings* section of a target. 
+
+# Next Steps
+
+Kotlin/Native has bidirectional interop with Objective-C and Swift languages. 
+Kotlin objects integrates with Objective-C/Swift reference counting. Unused Kotlin
+objects are automatically removed. You may want to see the detailed documentation on 
+the [Objective-C Interop](https://github.com/JetBrains/kotlin-native/blob/master/OBJC_INTEROP.md).
+Of course it is possible to import an existing Framework and use it from Kotlin. Kotlin/Native
+comes with a good set of pre-imported system Frameworks.
+
+Kotlin/Native supports C interop too. Checkout the [Kotlin/Native as a Dynamic Library](dynamic-libraries.html)
+tutorial for that, or have a look at the
+[C Interop](https://github.com/JetBrains/kotlin-native/blob/master/OBJC_INTEROP.md) documentation article
```

Comment:
Is this link correct?

Reply:
yep

Reply:
`OBJC_INTEROP.md` doesn't seem to cover anything related to C interop or `-produce dynamic`. What do I miss?

Reply:
fixed